### PR TITLE
Update Shelter Admin Fields (DEV-1003)

### PR DESCRIPTION
**Pull request by vecchp**
DEV-1003

Todo:
- [x] Handle other field
- [x] Add back media upload
- [x] Clear inconsistent entries choice enum updates
- [x] Use `django-select2` for multi-selects
- [x] Admin upload progress bar
- [x] S3 resumable storage (filesystem won't work in AWS) 
- [x] Add google places autocomplete
- [x] Create ticket to update s3 lifecycle to delete chunks older than 24 hours.
- [x] Add ability to add multiple contacts

https://github.com/user-attachments/assets/149348c6-2d7b-4f12-8da4-748bd6fdd36d

## Summary by Sourcery

Enhancements:
- Refactor shelter admin fields to include new categories such as demographics, special situation restrictions, room styles, and shelter programs.

## Summary by Sourcery

Update shelter admin fields to include new categories and improve the admin interface with dynamic field handling and resumable file uploads. Refactor models and enums to align with the new field categories and choices.

Enhancements:
- Refactor shelter admin fields to include new categories such as demographics, special situation restrictions, room styles, and shelter programs.
- Implement dynamic field handling in the admin interface to manage 'other' options for multiple-choice fields.
- Add support for resumable file uploads and progress tracking in the admin interface.

Build:
- Add 'admin_async_upload' and 'places' to the list of installed apps in the Django settings.
- Configure Google Maps and Places API settings for location fields.

Chores:
- Remove deprecated enums and replace them with updated choices for various fields.

### CODE REVIEW

**Comment by vecchp** [comment]
@sourcery-ai review

File: apps/betterangels-backend/common/models.py [hunk 1]
```diff
@@ -88,7 +89,8 @@ def save(self, *args: Any, **kwargs: Any) -> None:
             self.original_filename = self.file.name
 
             # Determine the MIME type of the file
-            mime_type = self.file.file.content_type
+            self.file.seek(0)
+            mime_type = magic.from_buffer(self.file.read(), mime=True)
```
Comment (vecchp) [inline apps/betterangels-backend/common/models.py:93]: Probably want to use magic only if `content_type` is not specified.

File: apps/betterangels-backend/common/admin.py [hunk 2]
```diff
@@ -72,6 +77,15 @@ class AttachmentAdmin(admin.ModelAdmin):
         "attachment_type",
     )
 
+    def formfield_for_dbfield(
```
Comment (vecchp) [inline apps/betterangels-backend/common/admin.py:80]: Maybe we don't want to modify attachment.

File: apps/betterangels-backend/shelters/admin.py [hunk 3]
```diff
@@ -2,118 +2,305 @@
...
+    )
+    storage = forms.MultipleChoiceField(
+        choices=StorageChoices,
+        widget=Select2MultipleWidget(
+            attrs={
+                "data-placeholder": "Select storage...",
+                "data-allow-clear": "true",
+            }
+        ),
+        required=True,
```
Comment (mikefeldberg) [inline apps/betterangels-backend/shelters/admin.py:118]: `storage`, `pets`, and `parking` aren't listed as required in the [source doc](https://docs.google.com/document/d/1dhPkGavRon-D2Ae4XJv1NUeYowJ3vl86npmI92ad4SU/edit?tab=t.0#heading=h.8amdd3r3eahw)

File: apps/betterangels-backend/shelters/models.py [hunk 4]
```diff
@@ -120,82 +145,130 @@ def __str__(self) -> str:
...
+    general_services = models.ManyToManyField(GeneralService)
+    health_services = models.ManyToManyField(HealthService)
+    training_services = models.ManyToManyField(TrainingService)
+    other_services = CKEditor5Field(null=True, blank=True)
+
+    # Entry Requirements
+    entry_info = CKEditor5Field(null=True, blank=True)
     entry_requirements = models.ManyToManyField(EntryRequirement)
+    bed_fees = CKEditor5Field(null=True, blank=True)
+    program_fees = CKEditor5Field(null=True, blank=True)
```
Comment (mikefeldberg) [inline apps/betterangels-backend/shelters/models.py:208]: I assume this was a judgement call over the stated "yes/no + short text" spec?

File: apps/betterangels-backend/shelters/models.py [hunk 5]
```diff
@@ -120,82 +145,130 @@ def __str__(self) -> str:
...
+
+    # Ecosystem Information
     cities = models.ManyToManyField(City)
-    city_district = models.PositiveSmallIntegerField(
-        validators=[MinValueValidator(1), MaxValueValidator(15)],
+    spa = models.ManyToManyField(SPA)
+    city_council_district = models.PositiveSmallIntegerField(
+        choices=[(i, str(i)) for i in range(1, 16)],
```
Comment (mikefeldberg) [inline apps/betterangels-backend/shelters/models.py:214]: slight preference for constants vs magic numbers

File: apps/betterangels-backend/shelters/models.py [hunk 6]
```diff
@@ -120,82 +145,130 @@ def __str__(self) -> str:
...
         null=True,
         blank=True,
-        verbose_name="LA City Council District (1-15)",
+        verbose_name="LA City Council District",
+        db_index=True,
     )
     supervisorial_district = models.PositiveSmallIntegerField(
-        validators=[MinValueValidator(1), MaxValueValidator(5)],
+        choices=[(i, str(i)) for i in range(1, 6)],
```
Comment (mikefeldberg) [inline apps/betterangels-backend/shelters/models.py:221]: same comment as above, re: magic numbers

**Review by mikefeldberg — APPROVED** [review]

**Pull request merged**